### PR TITLE
Added Min/Max/Exact units from a Course List Rules to Program

### LINKED ISSUE #255: Create/Edit Program Template Missing Course Rules
Same with #229, the Course (List) Rule needs extra rules. Course Rule needs:
Min/Exact/Max number of Units from a Course List.

### PR DESCRIPTION
**Pull request by spencerj411**
closes #255.

Same solution as on #229.

image

Edit:
added an error check for when nothing has been selected for the dropdown.

### CODE REVIEW

File: cassdegrees/templates/widgets/rulescripts.html
Comment (j-selby) [inline cassdegrees/templates/widgets/rulescripts.html:85]: Don't merge this one if it can be avoided - move this to a css class if needed.

File: cassdegrees/templates/widgets/rulescripts.html
Comment (j-selby) [inline cassdegrees/templates/widgets/rulescripts.html]: There doesn't appear to be error checking for this one - it can be saved with an empty value.

**Review by j-selby — CHANGES_REQUESTED** [review]
Few issues, but looks good. Note that this is likely going to break pre-existing rules, so planning a migration strategy might be useful here.

File: cassdegrees/templates/widgets/rulescripts.html
Comment (spencerj411) [reply cassdegrees/templates/widgets/rulescripts.html]: fixed with latest batch of commits

File: cassdegrees/templates/widgets/rulescripts.html
Comment (spencerj411) [reply cassdegrees/templates/widgets/rulescripts.html:85]: if I move it to style.css, it can't overwrite anu styling (`.anuform fieldset fieldset input` adds padding to the units input box). So I had to settle with the inline styling for the meantime. @JackSekoranja is aware of this decision of mine. But I agree, it 's not ideal.

**Review by Daniel-Jang — APPROVED** [review]
Works well, happy for you to merge it in once James' suggestions are addressed.

File: cassdegrees/static/css/style.css
Comment (j-selby) [inline cassdegrees/static/css/style.css]: `-` in css for consistency

**Review by j-selby — APPROVED** [review]
Looks good now otherwise!

File: cassdegrees/templates/widgets/rulescripts.html
Comment (j-selby) [inline cassdegrees/templates/widgets/rulescripts.html:86]: Sorry, didn't realise you had merged this yet - you can use !important to override css - e.g. `padding-left: 0 !important`, so you can still use classes for this one.

**Pull request merged**
